File: tests/test_nested_field_mutation.py
class Inner:
    def __init__(self):
        self.values: list[int] = []

class Outer:
    def __init__(self):
        self.inner: Inner = Inner()

class State:
    def __init__(self):
        self.data: Outer = Outer()

def mutate_nested_field(state: State, value: int) -> None:
    """Test that nested field mutation properly marks the variable as mutable."""
    # This pattern should generate: let mut obj = state.data.clone();
    obj = state.data
    obj.inner.values.append(value)

File: tests/test_nested_field_mutation.rs
#[derive(Debug, Clone)]
pub struct Inner {}
impl Inner {
    pub fn new() -> Self {
        Self {}
    }
}
#[derive(Debug, Clone)]
pub struct Outer {}
impl Outer {
    pub fn new() -> Self {
        Self {}
    }
}
#[derive(Debug, Clone)]
pub struct State {}
impl State {
    pub fn new() -> Self {
        Self {}
    }
}
#[doc = "Test that nested field mutation properly marks the variable as mutable."]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn mutate_nested_field(state: &State, value: i32) {
    let mut obj = state.data;
    obj.inner.values.push(value);
}
